paddle/fluid/operators/mkldnn/matmul_mkldnn_op.cc

```diff
@@ -80,6 +81,19 @@ class MatMulFactory {
     return dnnl::memory(md, engine_, to_void_cast(data));
   }
 
+  bool IsOutputFused(const ExecutionContext& ctx) const {
+    auto& fused_reshape_Out = ctx.Attr<std::vector<int>>("fused_reshape_Out");
+    auto& fused_transpose_Out =
+        ctx.Attr<std::vector<int>>("fused_transpose_Out");
+    return !fused_reshape_Out.empty() && !fused_transpose_Out.empty();
+  }
+
+  void correctStridesWhenOutputFused(const ExecutionContext& ctx,
+                                     const memory::dim N, memory::dim b,
+                                     memory::dims* out_strides) const {
+    if (IsOutputFused(ctx)) *out_strides = {N, b * N, 1};
```

Comment:
Please, correct me if I'm wrong, but I have an impression that the axis stored in `fused_transpose_Out` is currently completely disregarded in the code of dnnl's matmul, while as far I understand it, it influences how the strides are laid out. Perhaps it'd be advised to ensure with paddle enforce mechanism what transpose axis is currently supported.

Reply:
I see it used for shape_out 
https://github.com/PaddlePaddle/Paddle/pull/23866/files/d93bb88437fbbb690d11338e80f36707cec3750a#diff-126803a76a1b0cbf6198f30e23058f8dR419-R436

Reply:
It's used in the Op not the kernel

Reply:
That's right, thank you for clarification.